From 44dc46c133fa5e87d27eeef4c936e44cb886602b Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint.reczey@canonical.com>
Date: Wed, 26 Apr 2017 21:07:42 +0200
Subject: [PATCH 6/7] Show Ubuntu name on Ubuntu

Build depend on dpkg-dev (>= 1.15.1) and patch to do so.
---
 debian/control                        |  4 +++-
 debian/patches/ubuntu-installer.patch | 39 +++++++++++++++++++++++++++++++++++
 debian/rules                          | 14 +++++++++++++
 3 files changed, 56 insertions(+), 1 deletion(-)
 create mode 100644 debian/patches/ubuntu-installer.patch

diff --git a/debian/control b/debian/control
index d976102..9f9fa12 100644
--- a/debian/control
+++ b/debian/control
@@ -4,7 +4,9 @@ Priority: optional
 Maintainer: Debian Install System Team <debian-boot@lists.debian.org>
 Uploaders: Christian Perrier <bubulle@debian.org>
 Build-Depends: debhelper (>= 9)
-Build-Depends-Indep: dpkg-dev (>= 1.7.0), po-debconf (>= 0.5.0)
+Build-Depends-Indep: dpkg-dev (>= 1.15.1),
+                     patch,
+                     po-debconf (>= 0.5.0)
 Vcs-Browser: https://anonscm.debian.org/cgit/d-i/cdrom-detect.git
 Vcs-Git: https://anonscm.debian.org/git/d-i/cdrom-detect.git
 
diff --git a/debian/patches/ubuntu-installer.patch b/debian/patches/ubuntu-installer.patch
new file mode 100644
index 0000000..391b34b
--- /dev/null
+++ b/debian/patches/ubuntu-installer.patch
@@ -0,0 +1,39 @@
+diff --git a/debian/cdrom-detect.postinst b/debian/cdrom-detect.postinst
+index 0fbba9c..d5fea7c 100755
+--- a/debian/cdrom-detect.postinst
++++ b/debian/cdrom-detect.postinst
+@@ -28,7 +28,7 @@ try_mount() {
+ 			db_set cdrom-detect/cdrom_fs $type
+ 			ret=0
+ 		else
+-			log "The CD in $device is not a Debian CD!"
++			log "The CD in $device is not an Ubuntu CD!"
+ 			umount /cdrom 2>/dev/null || true
+ 			WRONG=1
+ 		fi
+@@ -94,7 +94,7 @@ fi
+ 
+ hw-detect cdrom-detect/detect_progress_title || true
+ 
+-log "Searching for Debian installation media..."
++log "Searching for Ubuntu installation media..."
+ 
+ mkdir /cdrom 2>/dev/null || true
+ 
+diff --git a/debian/cdrom-detect.templates b/debian/cdrom-detect.templates
+index 8da25d3..fe970dd 100644
+--- a/debian/cdrom-detect.templates
++++ b/debian/cdrom-detect.templates
+@@ -134,9 +134,10 @@ Description: for internal use; can be preseeded
+ 
+ Template: cdrom/suite
+ Type: select
+-Choices: stable, testing, unstable
++Choices: precise, trusty, xenial, yakkety, zesty
++Default: zesty
+ Description: for internal use only
+- Debian version to install
++ Ubuntu version to install
+ 
+ Template: cdrom/codename
+ Type: string
diff --git a/debian/rules b/debian/rules
index 78c7615..c4a7e1b 100755
--- a/debian/rules
+++ b/debian/rules
@@ -1,3 +1,17 @@
 #! /usr/bin/make -f
 %:
 	dh $@
+
+
+override_dh_auto_configure-indep:
+ifeq (yes,$(shell dpkg-vendor --derives-from Ubuntu && echo yes))
+	patch -p1 < $(CURDIR)/debian/patches/ubuntu-installer.patch
+endif
+	dh_auto_configure
+
+override_dh_auto_clean-indep:
+ifeq (yes,$(shell dpkg-vendor --derives-from Ubuntu && echo yes))
+	(patch -s -R -f -p1 --dry-run < $(CURDIR)/debian/patches/ubuntu-installer.patch > /dev/null && \
+		patch -R -p1 < $(CURDIR)/debian/patches/ubuntu-installer.patch) || true
+endif
+	dh_auto_clean
-- 
2.7.4

